l10n_th_tax_report/reports/tax_report.py

```diff
@@ -0,0 +1,113 @@
+            """
+            select company_id, account_id, partner_id,
+                tax_invoice_number, tax_date, name,
+                sum(tax_base_amount) tax_base_amount, sum(tax_amount) tax_amount
+            from (
+            select t.id, t.company_id, ml.account_id, t.partner_id,
+              case when ml.parent_state = 'posted' and t.reversing_id is null
+                then t.tax_invoice_number else
+                t.tax_invoice_number || ' (VOID)' end as tax_invoice_number,
+              t.tax_invoice_date as tax_date,
+              case when ml.parent_state = 'posted' and t.reversing_id is null
+                then t.tax_base_amount else 0.0 end as tax_base_amount,
+              case when ml.parent_state = 'posted' and t.reversing_id is null
+                then t.balance else 0.0 end as tax_amount,
+              case when m.ref is not null
+                then m.ref else ml.move_name end as name
+            from account_move_tax_invoice t
+              join account_move_line ml on ml.id = t.move_line_id
+              join account_move m on m.id = ml.move_id
+            where ml.parent_state in ('posted', 'cancel')
+              and t.tax_invoice_number is not null
+              and ml.account_id in (select distinct account_id
+                                    from account_tax_repartition_line
+                                    where account_id is not null
+                                    and invoice_tax_id = %s or refund_tax_id = %s)
+              and t.report_date >= %s and t.report_date <= %s
+              and ml.company_id = %s
+              and t.reversed_id is null
+            ) a
+            group by company_id, account_id, partner_id,
+                tax_invoice_number, tax_date, name
+            order by tax_date, tax_invoice_number
+        """,
+            (
+                self.tax_id.id,
+                self.tax_id.id,
+                self.date_from,
+                self.date_to,
+                self.company_id.id,
+            ),
+        )
+        tax_report_results = self._cr.dictfetchall()
+        ReportLine = self.env["tax.report.view"]
+        self.results = False
+        for line in tax_report_results:
+            self.results += ReportLine.new(line)
+
+    def print_report(self, report_type="qweb"):
+        self.ensure_one()
+        action = (
+            report_type == "xlsx"
+            and self.env.ref("l10n_th_tax_report.action_tax_report_xlsx")
+            or self.env.ref("l10n_th_tax_report.action_tax_report_pdf")
+        )
+        return action.report_action(self, config=False)
+
+    def _get_html(self):
+        result = {}
+        rcontext = {}
+        context = dict(self.env.context)
+        report = self.browse(context.get("active_id"))
+        if report:
+            rcontext["o"] = report
+            result["html"] = self.env.ref(
+                "l10n_th_tax_report.report_tax_report_html"
+            ).render(rcontext)
```

Comment:
```suggestion
            )._render(rcontext)
```


In core change to _render (view)